lib/src/working_copy.rs

```diff
@@ -69,9 +74,14 @@ pub struct FileState {
 }
 
 impl FileState {
+    #[cfg_attr(windows, allow(unused_variables))]
     fn for_file(executable: bool, size: u64, metadata: &Metadata) -> Self {
+        #[cfg(unix)]
+        let file_type = FileType::Normal { executable };
+        #[cfg(windows)]
+        let file_type = FileType::Normal { executable: () };
```

Comment:
It might be better to explicitly discard the unused variable here and elsewhere.

```suggestion
    fn for_file(executable: bool, size: u64, metadata: &Metadata) -> Self {
        #[cfg(unix)]
        let file_type = FileType::Normal { executable };
        #[cfg(windows)]
        let file_type = FileType::Normal { executable: {
            // Windows doesn't support executable bit.
            let _ = executable;
            ()
        } };
```

Reply:
Makes sense. Fixed here and in `write_path_to_store()`.